### PR TITLE
chore: migrate Breadcrumb component

### PR DESCRIPTION
**Pull request by SimonDegraeve**
## DESCRIPTION

Migrate the `Breadcrumb` component. There are no props change. I also extracted the mobile logic gradient in a hook.

## SCREENSHOTS / SCREEN RECORDINGS

image

## COMPATIBILITY

- [ ] Tested on Safari (desktop)
- [x] Tested on Chrome (desktop)
- [ ] Tested on Firefox (desktop)
- [ ] Tested on mobile device sizes
- [ ] Tested on tablet device sizes
- [ ] Tested on IOS Safari (either device or simulator)

## QA

- [x] Thoroughly tested in local environment
- [ ] Added tests for all new features
- [ ] Added tests that considered edge cases

### CODE REVIEW

File: lib/src/components/Breadcrumb/index.test.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import { render } from '../../../tests'
```
Comment (theo-mesnil) [inline lib/src/components/Breadcrumb/index.test.tsx]:
```suggestion
import { render } from '@tests'
```

File: lib/src/components/Breadcrumb/index.tsx [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+          {clones.reverse()}
+        </ol>
+
+        {endGradient}
+      </nav>
+    )
+  }
+)
+
+BreadcrumbComponent.displayName = 'Breadcrumb'
```
Comment (theo-mesnil) [inline lib/src/components/Breadcrumb/index.tsx]: Pas besoin à supprimer

File: lib/src/components/Breadcrumb/index.tsx [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react'
+
+import { primitives } from '@/theme/tokens'
+import { classNames } from '@/utils'
+import { RightIcon } from '@old/Icons'
+import { clamp, hexToRGBA, throttle } from '@old/utils'
```
Comment (theo-mesnil) [inline lib/src/components/Breadcrumb/index.tsx]: Il faut voir si on veut ajouter aux utils si on a en besoin,

File: lib/src/components/Breadcrumb/Item.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import { forwardRef } from 'react'
```
Comment (theo-mesnil) [inline lib/src/components/Breadcrumb/Item.tsx:1]: rename to Item.tsx

File: lib/src/components/Breadcrumb/types.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
+/* eslint-disable perfectionist/sort-modules */
+import type { ComponentPropsWithRef, HTMLAttributes } from 'react'
+
+import type { ThemeColorTokens } from '@old/theme'
```
Comment (theo-mesnil) [inline lib/src/components/Breadcrumb/types.ts]: Il ne faut plus utiliser cela, plus se dire que c'est du css ?

File: lib/src/components/Breadcrumb/breadcrumb.module.scss [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    right: 0;
+    transform-origin: right;
+  }
+
+  .startGradient,
+  .endGradient {
+    position: absolute;
+    bottom: 0;
+    top: 0;
+    width: 30;
```
Comment (cnairi) [inline lib/src/components/Breadcrumb/breadcrumb.module.scss]: 30 is not a valid value

File: lib/src/components/Breadcrumb/breadcrumb.module.scss [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    letter-spacing: var(--letter-spacing-subtitle-sm);
+    line-height: var(--line-height-subtitle-sm);
+    text-transform: uppercase;
+  }
+
+  .list {
+    display: inline-flex;
+    align-items: center;
+    max-width: 100%;
+    height: 100%;
```
Comment (cnairi) [inline lib/src/components/Breadcrumb/breadcrumb.module.scss]: height: 100% is already defined on root

File: lib/src/components/Breadcrumb/docs/examples/child-disabled.tsx [hunk 8]
```diff
@@ -0,0 +1,13 @@
+import { Breadcrumb } from '@old/Breadcrumb'
```
Comment (cnairi) [inline lib/src/components/Breadcrumb/docs/examples/child-disabled.tsx]: You shouldn't import the old Breadcrumb component in your doc

File: lib/src/components/Breadcrumb/docs/examples/separator.tsx [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+        <Breadcrumb.Item href="/">Components</Breadcrumb.Item>
+        <Breadcrumb.Item>Breadcrumb</Breadcrumb.Item>
+      </Breadcrumb>
+
+      <Breadcrumb separator="-">
+        <Breadcrumb.Item href="/">Introduction</Breadcrumb.Item>
+        <Breadcrumb.Item href="/">Components</Breadcrumb.Item>
+        <Breadcrumb.Item>Breadcrumb</Breadcrumb.Item>
+      </Breadcrumb>
+    </Stack>
```
Comment (cnairi) [inline lib/src/components/Breadcrumb/docs/examples/separator.tsx]: Stack should be replaced with <div className="flex"> as it's going to be deprecated

File: lib/src/components/Breadcrumb/index.tsx [hunk 10]
```diff
@@ -0,0 +1,170 @@
...
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react'
+
+import { primitives } from '@/theme/tokens'
+import { classNames } from '@/utils'
+import { RightIcon } from '@old/Icons'
```
Comment (cnairi) [inline lib/src/components/Breadcrumb/index.tsx]: Replace old RightIcon with the new one please `import { RightIcon } from '@/components/Icon'`

File: lib/src/components/Breadcrumb/index.tsx [hunk 11]
```diff
@@ -0,0 +1,170 @@
...
+import { Item } from './item'
+import type { BreadcrumbProps } from './types'
+
+const cx = classNames(breadcrumbStyles)
+
+export const BreadcrumbComponent = forwardRef<HTMLDivElement, BreadcrumbProps>(
+  (
+    {
+      children,
+      gradientBackground = 'neutral-10',
```
Comment (cnairi) [inline lib/src/components/Breadcrumb/index.tsx]: Could we remove the gradientBackground? It's only used in welcome-home

File: lib/src/components/Breadcrumb/index.tsx [hunk 12]
```diff
@@ -0,0 +1,170 @@
...
+      return cloneElement(child, {
+        isActive,
+        key: `breadcrumb-${index}`,
+        separator: isLastChild ? undefined : separator,
+        ...child.props,
+      })
+    })
+
+    // Add scroll gradient on mobile
+    const { endGradient, listRef, onListScroll, startGradient } = useGradient(gradientBackground)
```
Comment (cnairi) [inline lib/src/components/Breadcrumb/index.tsx]: I would completely get ridd off `gradientBackground`, @theo-mesnil what do you think? (only used in wh)

File: lib/src/components/Breadcrumb/item.tsx [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+import breadcrumbItemStyles from './item.module.scss'
+import type { BreadcrumbItemProps } from './types'
+
+const cx = classNames(breadcrumbItemStyles)
+
+/**
+ * @name Breadcrumb.Item
+ */
+export const Item = forwardRef<HTMLAnchorElement, BreadcrumbItemProps>(
+  ({ children, dataTestId, isActive, separator, ...rest }, ref) => {
```
Comment (cnairi) [inline lib/src/components/Breadcrumb/item.tsx]: I think we are trying to use 'data-testid' instead of dataTestId

File: lib/src/components/Breadcrumb/docs/examples/separator.tsx (hunk 9 above)
Comment (theo-mesnil) [reply lib/src/components/Breadcrumb/docs/examples/separator.tsx]: Use <div className /> instead

File: lib/src/components/Breadcrumb/index.tsx (hunk 12 above)
Comment (theo-mesnil) [reply lib/src/components/Breadcrumb/index.tsx]: Hum are you sure? We need to keep this on mobile i think

Comment (SimonDegraeve) [reply lib/src/components/Breadcrumb/index.tsx]: So, should I remove the gradient ?

**Comment by SimonDegraeve** [comment]
Thank you for the feedback, I will do the changes this morning

File: lib/src/components/Breadcrumb/index.tsx (hunk 12 above)
Comment (theo-mesnil) [reply lib/src/components/Breadcrumb/index.tsx]: For me now we need to keep it for mobile style, this can be rework on belonging follows the new design maybe

Comment (SimonDegraeve) [reply lib/src/components/Breadcrumb/index.tsx]: Maybe I can keep the gradient but remove the customizable color, what do you think ?

Comment (cnairi) [reply lib/src/components/Breadcrumb/index.tsx]: Yes, sorry I didn't realize you had a default value on useGradient, even without gradientBackground!!

File: lib/src/components/Breadcrumb/breadcrumb.module.scss [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+    transform-origin: right;
+  }
+
+  .startGradient,
+  .endGradient {
+    position: absolute;
+    bottom: 0;
+    top: 0;
+    width: var(--spacing-xxl);
+  }
```
Comment (theo-mesnil) [inline lib/src/components/Breadcrumb/breadcrumb.module.scss]: Do we want to use camelCase or snake-case ?
cc @cnairi
